src/Features/Lsif/Generator/Generator.cs

```diff
@@ -335,6 +374,33 @@ void MarkImplementationOfSymbol(ISymbol baseMember)
             return documentVertex.GetId();
         }
 
+        private static async Task GenerateSemanticTokensAsync(
+            Document document,
+            ILsifJsonWriter lsifJsonWriter,
+            IdFactory idFactory,
+            SourceText text,
+            LsifDocument documentVertex)
+        {
+            // Compute colorization data.
+            //
+            // Unlike the mainline LSP scenario, where we control both the syntatic colorizer (in-proc syntax tagger)
+            // and the semantic colorizer (LSP semantic tokens) LSIF is more likely to be consumed by clients
+            // which may have different syntatic classification behavior than us, resulting in missing colors. To avoid
+            // this, we include syntax tokens in the generated data.
+            var data = await SemanticTokensHelpers.ComputeSemanticTokensDataAsync(
+                document,
+                SemanticTokensHelpers.TokenTypeToIndex,
+                ProtocolConversions.TextSpanToRange(new TextSpan(0, text.Length), text),
```

Comment:
It's not a huge deal here, but we're converting a TextSpan to a Range, only to immediately convert it back to a text span again. But it looks like range can be null which already means "the entire file" so this could be simplified.

Reply:
Good catch! I didn't notice :)